networkx/algorithms/d_separation.py

```diff
@@ -356,15 +375,12 @@ def is_minimal_d_separator(G, u, v, z):
 
     https://en.wikipedia.org/wiki/Bayesian_network#d-separation
     """
-    if not nx.is_directed_acyclic_graph(G):
-        raise nx.NetworkXError("graph should be directed acyclic")
+    if not nx.d_separated(G, {u}, {v}, z):
+        return False
 
     union_uv = {u, v}
     union_uv.update(z)
```

Comment:
I think union_uv is no longer used. So we can remove this construction.

Reply:
Done